functions/Set-DbaAgReplica.ps1

```diff
@@ -154,6 +158,14 @@ function Set-DbaAgReplica {
                         $agreplica.SeedingMode = $SeedingMode
                     }
 
+                    if ($SessionTimeout) {
```

Comment:
This needs to be wrapped in  ShouldProess block for WhatIf support.

Reply:
Hi @wsmelton , thanks for reviewing this. The code I added was similar to the other properties and they are all wrapped in a ShouldProcess block:

            if ($Pscmdlet.ShouldProcess($server.Name, "Modifying replica for $($agreplica.Name) named $Name")) {
...
}

Is that okay or did you want an additional ShouldProcess block around the new section? Thanks.

Reply:
To me each one needs a should process instead of the overall action. If you pass in two parameters and only get one message for both it does not show you what it actually does.

So my opinion (may not be @potatoqualitee so will defer to her)...is remove the overall ShouldProcess and create one for each parameter to show what the command is actually going to change.

So instead of just:

```powershell
if ($Pscmdlet.ShouldProcess($server.Name, "Modifying replica for $($agreplica.Name) named $Name")) {
    ...
}
```

We do:
```powershell
server = $agreplica.Parent.Parent
                try {
                    if ($EndpointUrl) {
                        if ($PSCmdlet.ShouldProcess($server.Name, "Setting endpoint URL to $EndpointUrl for $($agreplica.Name)") {                        
                            $agreplica.EndpointUrl = $EndpointUrl
                        }
                    }

                    if ($FailoverMode) {
                        if ($PSCmdlet.ShouldProcess($server.Name, "Setting Failover Mode to $FailoverMode for $($agreplica.Name)") {                        
                            $agreplica.FailoverMode = [Microsoft.SqlServer.Management.Smo.AvailabilityReplicaFailoverMode]::$FailoverMode
                        }
                    }
...

Reply:
This would make it more closely represent what if you passed in any set of parameters.